pkg/providers/v1/aws.go

```diff
@@ -1194,7 +1195,16 @@ func init() {
 		}
 
 		var provider credentials.Provider
-		if cfg.Global.RoleARN == "" {
+		if os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE") != "" {
```

Comment:
I know we already have this unfortunate if statement, but it seems like the purpose of `credentials.NewChainCredentials` is to allow an order to be specified -- can we simply add the `NewWebIdentityRoleProvider` to the chain?  

I think what we should have done instead of `if cfg.Global.RoleARN == "" ...`, is to just create a custom provider similar to AssumeRoleProvider which takes an arn from the config file, and just add it to the NewChainCredentials call (if the role is not provided it should return an error and the [next provider will be called](https://github.com/aws/aws-sdk-go/blob/e2d6cb448883e4f4fcc5246650f89bde349041ec/aws/credentials/chain_provider.go#L74-L81)).  

Let me know if you already tried something like that--perhaps I'm misunderstanding why it was done this way.

Reply:
Actually, the identity session should have been picked up from the original session that we pass on with:
```
	&credentials.StaticProvider{
		Value: sess.Config.Credentials.Get(),
	}
```

Maybe it is as simple as enabling the shared config:
```
		sess, err = session.NewSessionWithOptions(session.Options{
			Config:            *cfg,
			SharedConfigState: session.SharedConfigEnable,
		})
```

Let me try that.